server/api/story/index.post.ts

```diff
@@ -1,49 +1,102 @@
-import gptscript from '@gptscript-ai/gptscript'
-import { Readable } from 'stream'
-import type { StreamEvent, StoryRequest } from '@/lib/types'
+import {CallFrame, RunFrame, Frame, GPTScript, RunEventType} from '@gptscript-ai/gptscript'
+import {Readable} from 'stream'
+import type {StoryRequest, StreamEvent} from '@/lib/types'
 
-export const runningScripts: Record<string, string>= {}
-export const eventStream = new Readable({read(){}});
+export const runningScripts: Record<string, string> = {}
+export const eventStream = new Readable({
+    read() {
+    }
+});
+const gptscript = new GPTScript()
 
 export default defineEventHandler(async (event) => {
     const request = await readBody(event) as StoryRequest
 
     // Ensure that the request has the required fields
-    if (!request.prompt) { throw createError({
-        statusCode: 400,
-        statusMessage: 'prompt is required'
-    })}
-    if (!request.pages) { throw createError({
-        statusCode: 400,
-        statusMessage: 'pages are required'
-    })}
+    if (!request.prompt) {
+        throw createError({
+            statusCode: 400,
+            statusMessage: 'prompt is required'
+        })
+    }
+    if (!request.pages) {
+        throw createError({
+            statusCode: 400,
+            statusMessage: 'pages are required'
+        })
+    }
+
+    console.log(`generating story for: ${JSON.stringify(request)}`)
 
     // Run the script with the given prompt and number of pages
     const {storiesVolumePath, scriptPath, gptscriptCachePath} = useRuntimeConfig()
-    const opts: { cacheDir?: string } = gptscriptCachePath ? { cacheDir: gptscriptCachePath } : {}
-    const {stdout, stderr, promise} = await gptscript.streamExecFile(
-        `${scriptPath}`, `--story ${request.prompt} --pages ${request.pages} --path ${storiesVolumePath}`, opts)
+    const opts = {
+        input: `--story ${request.prompt} --pages ${request.pages} --path ${storiesVolumePath}`,
+        credentialOverrides: ['sys.openai:OPENAI_API_KEY'],
```

Comment:
Is this necessary? I haven't seen this before.

Reply:
yes, we're not surfacing prompts, so it's necessary to override with the environment variables.